tensorflow/core/kernels/mkl_pooling_ops_common.cc

```diff
@@ -132,64 +156,93 @@ template class MklPoolingFwdPrimitive<bfloat16>;
 
 template <typename T>
 void MklPoolingBwdPrimitive<T>::Setup(const MklPoolingParams& bwdParams) {
-  DCHECK(bwdParams.alg_kind == pooling_max ||
-         bwdParams.alg_kind == pooling_avg ||
-         bwdParams.alg_kind == pooling_avg_include_padding ||
-         bwdParams.alg_kind == pooling_avg_exclude_padding)
+  DCHECK(bwdParams.alg_kind == ALGORITHM::pooling_max ||
+         bwdParams.alg_kind == ALGORITHM::pooling_avg ||
+         bwdParams.alg_kind == ALGORITHM::pooling_avg_include_padding ||
+         bwdParams.alg_kind == ALGORITHM::pooling_avg_exclude_padding)
       << "Pooling algorithm kind is not supported";
   context_.alg_kind = bwdParams.alg_kind;
 
-  // check whether it is 2d or 3d
-  bool is_2d = (bwdParams.dst_dims.size() == 4);
-  // Create memory desc
+  // Create memory descriptor.
   context_.diff_src_md.reset(new memory::desc(
       {bwdParams.src_dims}, MklDnnType<T>(), memory::format::any));
   context_.diff_dst_md.reset(new memory::desc(
       {bwdParams.dst_dims}, MklDnnType<T>(), bwdParams.src_format));
+
   context_.bwd_desc.reset(new pooling_backward::desc(
       bwdParams.alg_kind, *context_.diff_src_md, *context_.diff_dst_md,
       bwdParams.strides, bwdParams.filter_dims, bwdParams.padding_left,
       bwdParams.padding_right, padding_kind::zero));
 
-  // create a forward primitive,
-  // which will be used as a hint for creating backward primitive
+  // Create a forward primitive,
+  // which will be used as a hint for creating backward primitive.
   context_.fwd_desc.reset(new pooling_forward::desc(
       bwdParams.prop_kind, bwdParams.alg_kind, *context_.diff_src_md,
       *context_.diff_dst_md, bwdParams.strides, bwdParams.filter_dims,
       bwdParams.padding_left, bwdParams.padding_right, padding_kind::zero));
   context_.fwd_pd.reset(
-      new pooling_forward::primitive_desc(*context_.fwd_desc, cpu_engine));
+      new pooling_forward::primitive_desc(*context_.fwd_desc, cpu_engine_));
   context_.bwd_pd.reset(new pooling_backward::primitive_desc(
-      *context_.bwd_desc, cpu_engine, *context_.fwd_pd));
-
-  // store expected primitive format
-  context_.diff_src_fmt = static_cast<mkldnn::memory::format>(
-      context_.bwd_pd.get()->diff_src_primitive_desc().desc().data.format);
+      *context_.bwd_desc, cpu_engine_, *context_.fwd_pd));
+
+#ifdef ENABLE_MKLDNN_V1
+  // Store expected primitive format.
+  context_.diff_src_fmt = MEMORY_FORMAT::any;
```

Comment:
Are lines 190-191 necessary? They are already set in the constructor of the struct.